### PR TITLE
WIP: Simplify Makefile to build all addon subdirs

### PR DESCRIPTION
**Pull request by ninsbl**
Please test this modified example Makefile for raster addons in GRASS 7.
After the modification, all subdirs are listed and thus build by default, so addon-devs do not have to add new modules manually (and deleted / moved addons do not have to be removed from the Makefile). If it is an acceptable solution, Makefiles for vector imagery and so on can be adjusted accordingly.

It would however probably require to find a solution for "deprecated" addons.

Not sure if it works with all versions of make...

See: https://trac.osgeo.org/grass/ticket/3901

### CODE REVIEW

File: grass7/raster/Makefile [hunk 1]
```diff
@@ -13,130 +13,7 @@ OLD = \
 # unclear
```
Comment (neteler) [inline grass7/raster/Makefile]:
```suggestion
```

File: grass7/raster/Makefile [hunk 2]
```diff
@@ -13,130 +13,7 @@ OLD = \
...
 #	r.convert \
```
Comment (neteler) [inline grass7/raster/Makefile]:
```suggestion
```

**Comment by neteler** [comment]
The same works for the vector/Makefile etc., pls consider to expand this PR

**Comment by ninsbl** [comment]
> The same works for the vector/Makefile etc., pls consider to expand this PR

Yes, it would be the plan to apply this to the other AddOn types (vector, imagery, ...) as well.

Now I also committed a solution for filtering out deprecated AddOns; if not building those AddOns by default is the preferred solution. However, maybe it is better to add a DEPRECATED section to the manual instead (and have addons and manuals built)?

**Comment by neteler** [comment]
Please also update misc/Makefile, then it should be complete (and there is a conflict in grass7/raster/Makefile)

**Pull request merged**